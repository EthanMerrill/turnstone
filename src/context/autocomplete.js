import React, { createContext, useState, useEffect } from 'react'

const AutocompleteContext = createContext()

const AutocompleteContextProvider = (props) => {
  const { children, splitChar, styles = {}, text = '' } = props
  const [queryState, setQueryState] = useState(text)
  const [highlightedState, setHighlightedState] = useState()
  const [selectedState, setSelectedState] = useState()
  const [customStyles] = useState(styles)
  const [splitCharState] = useState(splitChar)

  useEffect(() => setQueryState(text), [text])

  return (
    <AutocompleteContext.Provider
      value={{
        queryState,
        setQueryState,
        highlightedState,
        setHighlightedState,
        selectedState,
        setSelectedState,
        customStyles,
        splitCharState
      }}>
      {children}
    </AutocompleteContext.Provider>
  )
}

export { AutocompleteContext, AutocompleteContextProvider }
