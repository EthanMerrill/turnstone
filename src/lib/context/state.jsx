import React, { createContext, useReducer, useEffect } from 'react'
import reducer from '../reducers/reducer'
import {setQuery} from '../actions/actions'
import undef from '../utils/undef'

const StateContext = createContext()

const StateContextProvider = (props) => {
  const { styles = {}, text = '', items = [] } = props
  const { children, ...propsMinusChildren} = props
  const [state, dispatch] = useReducer(reducer, {
    query: text,
    items,
    itemsError: false,
    canShowListbox: false,
    highlighted: undef,
    selected: undef,
    customStyles: styles,
    props: propsMinusChildren
  })

  useEffect(() => dispatch(setQuery(text)), [text]) // TODO: Is this needed?

  return (
    <StateContext.Provider
      value={{state, dispatch}}>
      {children}
    </StateContext.Provider>
  )
}

export { StateContext, StateContextProvider }
