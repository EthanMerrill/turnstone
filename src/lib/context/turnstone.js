import React, { createContext, useState, useEffect } from 'react'

const TurnstoneContext = createContext()

const TurnstoneContextProvider = (props) => {
  const { children, splitChar, styles = {}, text = '' } = props
  const [queryState, setQueryState] = useState(text)
  const [highlightedState, setHighlightedState] = useState()
  const [selectedState, setSelectedState] = useState()
  const [customStyles] = useState(styles)
  const [splitCharState] = useState(splitChar)

  useEffect(() => setQueryState(text), [text])

  return (
    <TurnstoneContext.Provider
      value={{
        queryState,
        setQueryState,
        highlightedState,
        setHighlightedState,
        selectedState,
        setSelectedState,
        customStyles,
        splitCharState
      }}>
      {children}
    </TurnstoneContext.Provider>
  )
}

export { TurnstoneContext, TurnstoneContextProvider }
