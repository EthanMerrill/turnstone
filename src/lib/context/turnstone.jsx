import React, { createContext, useReducer, useState, useEffect } from 'react'
import reducer from '../reducers/reducer'
import {setQuery} from '../actions/actions'
import undef from '../utils/undef'

const TurnstoneContext = createContext() //TODO: Rename GlobalStateContext

const TurnstoneContextProvider = (props) => {
  const { children, splitChar, styles = {}, text = '', items = [] } = props
  const [state, dispatch] = useReducer(reducer, {
    query: text,
    items,
    highlighted: undef,
    selected: undef,
    customStyles: styles,
    splitChar
  })
  const [selectedState, setSelectedState] = useState()
  const [customStyles] = useState(styles)
  const [splitCharState] = useState(splitChar)

  useEffect(() => dispatch(setQuery(text)), [text]) // TODO: Is this needed?

  return (
    <TurnstoneContext.Provider
      value={{
        state,
        dispatch,
        selectedState,
        setSelectedState,
        customStyles,
        splitCharState
      }}>
      {children}
    </TurnstoneContext.Provider>
  )
}

export { TurnstoneContext, TurnstoneContextProvider }
