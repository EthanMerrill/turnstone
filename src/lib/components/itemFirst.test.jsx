import React from 'react'
import renderer from 'react-test-renderer'

import { vi, describe, expect, test } from 'vitest'
import { StateContextProvider } from '../context/state'
import ItemFirst from './itemFirst'

vi.mock('./item', () => ({ default: () => 'Item' }))

describe('ItemFirst', () => {
  test('Component renders correctly', () => {
    const customStyles = {
      groupName: 'group-name-class'
    }

    const component = renderer.create(
      <StateContextProvider styles={customStyles}>
       <ItemFirst
          groupName={'Cities'}
          index={0}
          key={'item0'}
        />
      </StateContextProvider>
    )

    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })
})