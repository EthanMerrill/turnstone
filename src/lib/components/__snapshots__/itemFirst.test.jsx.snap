// Vitest Snapshot v1

exports[`ItemFirst > Component renders correctly 1`] = `
[
  <div
    className="group-heading-class"
  >
    Cities
  </div>,
  "Item",
]
`;
